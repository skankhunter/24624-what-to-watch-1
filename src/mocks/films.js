const films = [
  {
    id: `1`,
    title: `John Wick`,
    genre: [`Crime`],
    poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: `2`,
    title: `Killers Bodyguard`,
    genre: [`Crime`, `Comedies`],
    poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    id: `3`,
    title: `Star Wars`,
    genre: [`Sci-Fi`],
    poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  },
  {
    id: `4`,
    title: `The Grand Budapest Hotel`,
    genre: [`Dramas`, `Comedies`],
    poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    src: `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  },
  {
    id: `5`,
    title: `The Cabin in the Woods`,
    genre: [`Thrillers`],
    poster: `fantastic-beasts-the-crimes-of-grindelwald.jpg`,
    src: `https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4`
  }
];

export default films;
