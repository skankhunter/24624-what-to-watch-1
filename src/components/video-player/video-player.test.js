import React from 'react';
import renderer from 'react-test-renderer';

import VideoPlayer from './video-player';
import films from '../../mocks/films';

describe(`The application is displayed correctly.`, () => {
  it(`VideoPlayer correctly renders after launch`, () => {
    const {src, poster} = films[0];

    const component = renderer.create(
        <VideoPlayer
          src={src}
          poster={poster}
          muted={true}
          isPlaying={false}
        />,
        {
          createNodeMock: () => {
            return {};
          }
        }
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
