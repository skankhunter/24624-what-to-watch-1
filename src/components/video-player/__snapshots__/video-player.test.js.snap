// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The application is displayed correctly. VideoPlayer correctly renders after launch 1`] = `
<video
  muted={true}
  poster="fantastic-beasts-the-crimes-of-grindelwald.jpg"
  src="https://download.blender.org/durian/trailer/sintel_trailer-480p.mp4"
/>
`;
